NitroxPatcher/Main.cs

```diff
@@ -28,16 +32,34 @@ public static void Execute()
         {
             Log.Setup(true);
             Optional.ApplyHasValueCondition<Object>(o => (bool)o);
+            Log.Info($"Using Nitrox version {Assembly.GetExecutingAssembly().GetName().Version} built on {File.GetCreationTimeUtc(Assembly.GetExecutingAssembly().Location)}");
 
             if (container != null)
             {
-                Log.Warn($"Patches have already been detected! Call {nameof(Apply)} or {nameof(Restore)} instead.");
+                Log.Error($"Patches have already been detected! Call {nameof(Apply)} or {nameof(Restore)} instead.");
                 return;
             }
 
             Log.Info("Registering dependencies");
             container = CreatePatchingContainer();
-            NitroxServiceLocator.InitializeDependencyContainer(new ClientAutoFacRegistrar());
+            try
+            {
+                NitroxServiceLocator.InitializeDependencyContainer(new ClientAutoFacRegistrar());
+            }
+            catch (ReflectionTypeLoadException ex)
+            {
+                Log.Error($"Failed to load one or more dependency types for Nitrox. Assembly: {ex.Types.FirstOrDefault()?.Assembly.FullName ?? "unknown"}");
+                foreach (Exception loaderEx in ex.LoaderExceptions)
+                {
+                    Log.Error(loaderEx);
+                }
+                throw;
+            }
+            catch (Exception)
+            {
+                Log.Error("Error while initializing and loading dependencies.");
```

Comment:
```suggestion
            catch (Exception ex)
            {
                Log.Error(ex, "Error while initializing and loading dependencies.");
```

Use `public static void Error(Exception ex, string message) => logger.Error(ex, message);` method variant.